### ISSUE TITLE
Bug - Unable to run single e2e test after pr #3827

### ISSUE THREAD

**Issue by bamsargeant** [ionic-team/stencil#3874]
# Bug
Unable to run a single e2e test

## Details
Unable to run single e2e test after upgrading from _v2.16.1_ to _v2.19.3_

Command used: `npx stencil test --e2e dialog.e2e.ts`

## Work around
I am able to get the same functionality by adding `--` as an extra arg

Command used: `npx stencil test --e2e -- dialog.e2e.ts`

### v2.16.1
![image](https://user-images.githubusercontent.com/8199374/205779216-03966fd1-452c-4e9d-9642-81c38048b817.png)

### v2.19.3
![image](https://user-images.githubusercontent.com/8199374/205779088-7b4d28dd-217f-4f62-a74c-a85a3b98ba53.png)

### v2.19.3 with workaround
![image](https://user-images.githubusercontent.com/8199374/205779425-2dc0b7ec-852d-422a-a915-925aa249cfbd.png)

_Originally posted by @bamsargeant in https://github.com/ionic-team/stencil/issues/3827#issuecomment-1338509558_

**Comment by rwaskiewicz** [comment]
Hey @bamsargeant 👋 

I'm trying to reproduce this issue, but need a little more information from you. PR #3827 isn't a part of Stencil [v2.19.3](https://github.com/ionic-team/stencil/releases/tag/v2.19.3). Instead, it was released as a part of[ Stencil v2.20.0](https://github.com/ionic-team/stencil/releases/tag/v2.20.0) (which was released just yesterday). This makes me think that your issue might be resolved by upgrading to v2.20.0.

I took a quick look by spinning up a minimal reproduction case by running `npx init stencil component jest-e2e-test` in my terminal. I used `npm run generate` to create `another-cmp` and `dialog-cmp` components, and attempted to reproduce this with both v2.19.0 and v2.20.0.

My directory structure looks like this:
```
➜  jest-e2e-test tree src/components/
src/components/
├── another-cmp
│   ├── another-cmp.css
│   ├── another-cmp.tsx
│   └── test
│       ├── another-cmp.e2e.ts
│       └── another-cmp.spec.tsx
├── dialog-cmp
│   ├── dialog-cmp.css
│   ├── dialog-cmp.tsx
│   └── test
│       ├── dialog-cmp.e2e.ts
│       └── dialog-cmp.spec.tsx
└── my-component
    ├── my-component.css
    ├── my-component.e2e.ts
    ├── my-component.spec.ts
    ├── my-component.tsx
    └── readme.md

5 directories, 13 files
```

With v2.19.0, I can reproduce:
```
➜  jest-e2e-test npm ls @stencil/core
jest-e2e-test@0.0.1 /private/tmp/jest-e2e-test
└── @stencil/core@2.19.3

➜  jest-e2e-test npx stencil test --e2e dialog-cmp.e2e.ts
[40:47.8]  @stencil/core
[40:48.0]  v2.19.3 🌏
[40:48.0]  testing e2e files
[40:48.8]  build, jest-e2e-test, dev mode, started ...
[40:48.9]  transpile started ...
[40:51.1]  transpile finished in 2.23 s
[40:51.1]  copy started ...
[40:51.1]  generate custom elements started ...
[40:51.1]  generate lazy started ...
[40:51.2]  copy finished (0 files) in 104 ms
[40:51.4]  generate custom elements finished in 300 ms
[40:51.4]  generate lazy finished in 298 ms
[40:51.4]  build finished in 2.62 s

[40:51.5]  jest args: --e2e dialog-cmp.e2e.ts --max-workers=8
 PASS  src/components/dialog-cmp/test/dialog-cmp.e2e.ts
 PASS  src/components/my-component/my-component.e2e.ts
 PASS  src/components/another-cmp/test/another-cmp.e2e.ts

Test Suites: 3 passed, 3 of 7 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        1.668 s
Ran all test suites.
```
With v2.20.0, I cannot reproduce:
```
➜  jest-e2e-test npx stencil test --e2e dialog-cmp.e2e.ts
[41:06.9]  @stencil/core
[41:07.0]  v2.20.0 🍁
[41:07.0]  testing e2e files
[41:07.8]  build, jest-e2e-test, dev mode, started ...
[41:07.8]  transpile started ...
[41:08.9]  transpile finished in 1.02 s
[41:08.9]  copy started ...
[41:08.9]  generate custom elements started ...
[41:08.9]  generate lazy started ...
[41:09.0]  copy finished (0 files) in 99 ms
[41:09.2]  generate custom elements finished in 292 ms
[41:09.2]  generate lazy finished in 291 ms
[41:09.2]  build finished in 1.39 s

[41:09.2]  jest args: --e2e --max-workers=8 dialog-cmp.e2e.ts
 PASS  src/components/dialog-cmp/test/dialog-cmp.e2e.ts
  dialog-cmp
    ✓ renders (195 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        0.419 s, estimated 1 s
Ran all test suites matching /dialog-cmp.e2e.ts/i.
```

Can you try upgrading to v2.20.0 and see if that fixes it? If not, can you please create a minimal reproduction case for the team to take a look at?

**Comment by bamsargeant** [comment]
@rwaskiewicz Perfecto! 

v2.20.0 fixed the issue - many thanks :)

Closing the ticket.
